libs/SalesforceSDKCore/SalesforceSDKCore/Classes/OAuth/SFOAuthCoordinator.m

```diff
@@ -811,8 +823,26 @@ - (NSURLSession*)session {
     return _session;
 }
 
+- (void)handleCustomDomainUpdateWithLoginHint:(NSString *)loginHint myDomain:(NSString *)myDomain {
+    self.domainUpdated = YES;
+    [self stopAuthentication];
+    self.loginHint = loginHint;
+    self.credentials.domain = myDomain;
+    [[SFUserAccountManager sharedInstance] setLoginHost:myDomain];
+    [self authenticate];
+}
+
 #pragma mark - WKNavigationDelegate (User-Agent Token Flow)
 - (void)webView:(WKWebView *)webView decidePolicyForNavigationAction:(WKNavigationAction *)navigationAction decisionHandler:(void (^)(WKNavigationActionPolicy))decisionHandler {
+    
+    SFDomainDiscoveryResult *result = [self.domainDiscoveryCoordinator handleWithWebAction:navigationAction];
```

Comment:
Maybe a little comment.
Should it be part of the `if() else if()` etc that follows?
```Objective-C
NSURL *url = navigationAction.request.URL;
NSString *requestUrl = [url absoluteString];
SFDomainDiscoveryResult *discoveryResult = [self.domainDiscoveryCoordinator handleWithWebAction:navigationAction];
if (discoveryResult) {
...
} else if (self isRedirectURL:requestUrl]) {
...
} else if ...
```